src/http.rs

```diff
@@ -47,10 +53,15 @@ where
 {
 	Normal(T),
 	NotFound,
+	NotFinalized,
 	Error(anyhow::Error),
 }
 
-fn confidence(block_num: u64, db: Arc<DB>) -> ClientResponse<ConfidenceResponse> {
+fn confidence(
+	block_num: u64,
+	db: Arc<DB>,
+	counter: Arc<Mutex<u64>>,
```

Comment:
Is there an option to pass u64 value to confidence and latest_block methods, and read counter before fn call?